interfaces/builtin/x11.go

```diff
@@ -138,6 +139,13 @@ owner /run/user/[0-9]*/mutter/Xauthority r,
 network netlink raw,
 /run/udev/data/c13:[0-9]* r,
 /run/udev/data/+input:* r,
+
+# Deny access to ICE granted by abstractions/X
+# See: https://bugs.launchpad.net/snapd/+bug/1901489
+deny owner @{HOME}/.ICEauthority r,
```

Comment:
should this be just `deny ${HOME}/...` ? I'm not familiar with how `deny` interacts with `owner`? The only other place I see for `deny owner` is in a commented out snippet from Jamie 4 years ago in the common template, and in tests.

Reply:
`deny owner @{HOME}/.ICEauthority r,` is fine - `owner` says that to allow `r` to only files that are owned by the accessing user (since `@{HOME}` is just a regex which provides access to all `/home` not `/home/$USER` since apparmor doesn't know the real `$HOME` so by using `owner` in the original rule in the X abstraction this ensures the user is only allowed access to their own file as they are the owner of that)  - and `deny` says to make this denied (so that we override the existing allowed access provided by the X abstraction).

Reply:
You can see examples of existing `deny owner` combinations at say https://gitlab.com/apparmor/apparmor/-/wikis/Profiling_with_tools - or even `audit deny owner` in `/etc/apparmor.d/abstractions/ubuntu-helpers` on Ubuntu systems.

Reply:
I wasn't sure myself so I went with the simplest initial choice, just deny the originally granted rule. Based on Alex's description it seem this is fine, though I would not mind reducing this to just `deny ...`, that is without the owner field.

Reply:
The `deny owner` is fine, I just wanted clarity on that, I didn't find the example upstream, thanks @alexmurray for the link.

Reply:
Sorry about this being after the fact, but I think that 'deny' would've been better than 'deny owner' since you're trying to undo something from somewhere else and I'd just assume block accesses to all ICEauthority files, whether they are owned by the user or not. What is committed is not 'wrong' since the X abstraction only allows owner match, but if something else allowed access to all ICEauthority files, then the 'deny owner' rule would allow reads to other users' ICEauthority files, only denying the ones owned by the user.